cmd/nfd-topology-updater/main.go

```diff
@@ -140,6 +146,7 @@ func initFlags(flagset *flag.FlagSet) (*topology.Args, *resourcemonitor.Args) {
 	flagset.StringVar(&args.ConfigFile, "config", "/etc/kubernetes/node-feature-discovery/nfd-topology-updater.conf",
 		"Config file to use.")
 	flagset.BoolVar(&resourcemonitorArgs.PodSetFingerprint, "pods-fingerprint", false, "Compute and report the pod set fingerprint")
+	flagset.StringVar(&args.KubeletDirPath, "kubelet-dir-path", DefaultKubeletDirPath, "Kubelet directory path for watching state and checkpoint files. passing an empty string disable the watching")
```

Comment:
Now I started to think about the naming. Would `-kubelet-state-dir` be better, more descriptive. WDYT?

Reply:
Sure, I can change it. but then it would be less descriptive wrt kubelet config file location https://github.com/kubernetes-sigs/node-feature-discovery/pull/1031#discussion_r1129172645.

Reply:
IMO, "kubelet dir" is very ambiguous, there can be many different "kubelet dirs" around the system. This is specifically about the kubelet state directory. Any thoughts @ffromani @PiotrProkop ?

Reply:
I agree with you @marquiz . We need to watch the directory(ies) on which the resource manager state file are sitting. We may even want to have multiple --kubelet-state-dirs in general.

Reply:
If I understand correctly the `KubeletDirPath` should correspond to Kubelet's `--root-dir` arg. Right now the files that are being watched are hardcoded to those 3:
```
"cpu_manager_state",
"memory_manager_state",
"kubelet_internal_checkpoint",
```

So in my opinion we should either change this flat to `--kubeler-root-dir` or allow specifying multiple kubelet checkpoint files, e.g:
```
--kubelet-state-file /var/lib/kubelet/cpu_manager_state --kubelet-state-file /var/lib/kubelet/memory_manager_state --kubelet-state-file /var/lib/kubelet/kubelet_internal_checkpoint
```
this will allow us to be more flexible ? WDYT?

Reply:
I like the flexibility and the general idea, but because the way kubelet updates the state files (tempfile + full rewrite + rename vs incremental writes) it's much safer to watch the parent directory rather than the state files

Reply:
> I like the flexibility and the general idea, but because the way kubelet updates the state files (tempfile + full rewrite + rename vs incremental writes) it's much safer to watch the parent directory rather than the state files

`pkg/utils/fswatcher.go` would handle renames and everything but that's probably an overkill here. Just watching the parent dir sounds good to me

Reply:
> > I like the flexibility and the general idea, but because the way kubelet updates the state files (tempfile + full rewrite + rename vs incremental writes) it's much safer to watch the parent directory rather than the state files
> 
> `pkg/utils/fswatcher.go` would handle renames and everything but that's probably an overkill here. Just watching the parent dir sounds good to me

good to know! To clarify, in my previous statement I was referring to my knowledge about the 3rd party package `fsnotify`.

Reply:
> pkg/utils/fswatcher.go would handle renames and everything but that's probably an overkill here. Just watching the parent dir sounds good to me

Nice! 

Having a list of separated files makes it more complex but more secure.

Reply:
> I was referring to my knowledge about the 3rd party package `fsnotify`.

Yeah, I got that. I wrote the nfd fswatcher to come over those limitations (even though it's probably now excessively bulletproof handling renames and all changes in all parent directories, too...)

Reply:
> Having a list of separated files makes it more complex but more secure, since we're avoiding mount the whole directory.

The fswatcher is just really simple-and-stupid (and thus reliable) as it always re-initializes all fsnotify watches if any changes are detected:
https://github.com/kubernetes-sigs/node-feature-discovery/blob/9e55028c92c5c48914fe496a8d159366af78e564/pkg/utils/fswatcher.go#L150

It was written with the config file in mind that is not supposed to update that frequently.

So in case of kubelet state files it might cause quite a bit of churn on deletion/addition of fsnotify watches

Reply:
As agreed, I changed the name to `-kubelet-state-dir` and kept it as a directory, so the watcher will watch the directory as a whole.
We can reevaluate in the future if specifying multiple separate checkpoint files is needed.

Reply:
A bit more bike-shedding on the naming 🙈 @PiotrProkop @ffromani are you ok with `-kubelet-state-dir` or would you prefer `-kubelet-state-dir`(what @PiotrProkop suggested)? I'm ok with either one

> We can reevaluate in the future if specifying multiple separate checkpoint files is needed.

👍

Reply:
Either one is good for me :)